### ISSUE TITLE
Eventflow output to ELK with log4net input and NDC stack

### ISSUE THREAD

**Issue by teeboy75** [Azure/diagnostics-eventflow#287]
I have a simple eventflow setup writing to elasticsearch with log4net input

`class Program
    {
        static void Main(string[] args)
        {
            using (var pipeline = DiagnosticPipelineFactory.CreatePipeline(".\\eventFlowConfig.json"))
            {
                var logger = LogManager.GetLogger("EventFlowRepo", "MY_LOGGER_NAME");
                using (LogicalThreadContext.Stacks["NDC"].Push("12345"))
                {
                    logger.Debug("Hey! Listen!", new Exception("uhoh"));
                }
            }

            Console.ReadLine();
        }
    }
`
eventflowconfig.json
`{
  "inputs": [    
    {
      "type": "Log4net",
      "logLevel": "Debug"
    },
    {
      "type": "Log4net",
      "logLevel": "Info"
    },
    {
      "type": "Log4net",
      "logLevel": "Warn"
    },
    {
      "type": "Log4net",
      "logLevel": "Error"
    },
    {
      "type": "Log4net",
      "logLevel": "Fatal"
    }
  ],  
  "outputs": [
    {
      "type": "ElasticSearch",
      "indexNamePrefix": "defaultindex",
      "serviceUri": "http://servername:9200",
      "basicAuthenticationUserName": "user",
      "basicAuthenticationUserPassword": "password",
      "eventDocumentTypeName": "diagData",
      "numberOfShards": 1,
      "numberOfReplicas": 0,
      "refreshInterval": "5s"     
    }   
  ],
  "settings": {
    "pipelineBufferSize": "1000",
    "maxEventBatchSize": "100",
    "maxBatchDelayMsec": "500",
    "maxConcurrency": "8",
    "pipelineCompletionTimeoutMsec": "30000"
  },
  "extensions": []
}
`

The issue is, the custom data I write to the NDC context, is not written to elasticsearch. How do I configure elastic output to recognize any custom data pushed to NDC?

**Comment by teeboy75** [comment]
Just to be clear, if I remove eventflow input and output and directly use log4nets log appenders, the NDC comment works..

**Comment by karolz-ms** [comment]
OK, thanks. I looked a bit more closely into this and it looks like appending data from Log4net global context & logical thread context was never implemented.

For comparison here is [how Stackify Retrace appender does this](https://github.com/stackify/stackify-api-dotnet/blob/master/Src/StackifyLib.log4net/StackifyAppender.cs#L282)

@teeboy75 would you be willing to submit a PR to have these properties added?

CC @jeremysmith1

**Comment by karolz-ms** [comment]
Thank you! Much appreciate your help @teeboy75

**Comment by teeboy75** [comment]
@karolz-ms 
I have the code ready. Waiting from corporate to give me permission for PR submission. will update you.

**Comment by karolz-ms** [comment]
@teeboy75 Roger!

**Comment by karolz-ms** [comment]
@pattisapu01 Do you mind if we (myself or some other Microsoft developer) "steal" the code from you (https://github.com/pattisapu01/diagnostics-eventflow/commit/c9c4d65cc20cd2ce61724413f35cb9c0de6f3741) and finish this PR? I'd be happy if your time and  work on this issue did not go to waste...

**Comment by pattisapu01** [comment]
Absolutely! Unfortunately, the company I work for is not moving quickly. Thank you for getting back to me.
Regards
Prakash

Get Outlook for Android<https://aka.ms/ghei36>

________________________________
